bento/src/main/java/com/yelp/android/bento/componentcontrollers/ViewPager2ComponentController.java

```diff
@@ -0,0 +1,292 @@
+    }
+
+    @NonNull
+    @Override
+    public ComponentController addComponent(int index, @NonNull Component component) {
+        mComponentGroup.addComponent(index, component);
+        return this;
+    }
+
+    @NonNull
+    @Override
+    public ComponentController addComponent(int index, @NonNull ComponentGroup componentGroup) {
+        mComponentGroup.addComponent(index, componentGroup);
+        return this;
+    }
+
+    @NonNull
+    @Override
+    public ComponentController addAll(@NonNull Collection<? extends Component> components) {
+        mComponentGroup.addAll(components);
+        return this;
+    }
+
+    @NonNull
+    @Override
+    public ComponentController replaceComponent(int index, @NonNull Component component) {
+        mComponentGroup.replaceComponent(index, component);
+        return this;
+    }
+
+    @NonNull
+    @Override
+    public ComponentController replaceComponent(int index, @NonNull ComponentGroup componentGroup) {
+        mComponentGroup.replaceComponent(index, componentGroup);
+        return this;
+    }
+
+    @NonNull
+    @Override
+    public Component remove(int index) {
+        return mComponentGroup.remove(index);
+    }
+
+    @Override
+    public boolean remove(@NonNull Component component) {
+        return mComponentGroup.remove(component);
+    }
+
+    @Override
+    public void clear() {
+        mComponentGroup.clear();
+    }
+
+    @Override
+    public void scrollToComponent(@NonNull Component component, boolean smoothScroll) {
+        // We need to figure out which page the component belongs to. A component may be a page or
+        // within a page, so we iterate through all the pages and check if the component exists on
+        // that page.
+        for (int i = 0; i < getSize(); i++) {
+            Component page = get(i);
+            if (page == component
+                    || (page instanceof ComponentGroup
+                    && ((ComponentGroup) page).findComponentOffset(component) != -1)) {
+                mViewPager.setCurrentItem(i, smoothScroll);
+            }
+        }
+    }
+
+    /**
+     * Not supported. Use {@link #scrollToComponent(Component, boolean)} instead.
+     */
+    @Override
+    public void scrollToComponentWithOffset(@NonNull Component component, int offset) {
+        throw new UnsupportedOperationException(
+                "Scrolling with offset is not supported for "
+                        + "ViewPagerComponent. Use scrollToComponent(Component, boolean) instead.");
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        return getViewTypeFromComponent(position);
+    }
+
+    /**
+     * Gets a view type from a {@link Component} for the specified position. This will return a
+     * unique integer for each unique view type and the same integer for the same view types.
+     */
+    @SuppressWarnings("unchecked") // Unchecked Component generics.
+    private int getViewTypeFromComponent(int position) {
```

Comment:
This is copied from `RecyclerViewComponentController` ( i cant subclass it because it uses recyclerview and we want viewpager2)